pkg/cableengine/syncer/syncer_test.go

```diff
@@ -335,9 +342,22 @@ func (t *testDriver) run() {
 		t.handledError <- err
 	})
 
+	Expect(submarinerv1.AddToScheme(kubeScheme.Scheme)).To(Succeed())
+
+	scheme := runtime.NewScheme()
+	Expect(submarinerv1.AddToScheme(scheme)).To(Succeed())
+
+	dynamicClient := fakeClient.NewSimpleDynamicClient(scheme)
+	restMapper := test.GetRESTMapperFor(&submarinerv1.Endpoint{})
+
 	client := fakeClientset.NewSimpleClientset()
 	t.gateways.GatewayInterface = client.SubmarinerV1().Gateways(namespace)
-	t.syncer = syncer.NewGatewaySyncer(t.engine, t.gateways, t.expectedGateway.Status.Version)
+	t.healthChecker, _ = healthchecker.New(&watcher.Config{
+		RestMapper: restMapper,
+		Client:     dynamicClient,
+		Scheme:     scheme,
+	}, namespace, "west")
```

Comment:
I don't think it makes sense to use the real health checker implementation. We should use a fake one so we can test  the interaction with it.  But we can leave this as is for now - I can follow up with another PR.

Reply:
ok , thanks . :)